src/web/mjs/connectors/ReaperScans.mjs

```diff
@@ -27,80 +28,71 @@ export default class ReaperScans extends Connector {
             return{
                 id: this.getRootRelativeOrAbsoluteLink(element, request.url),
                 title : element.text.trim()
             };
         });
     }
+
     async _getChapters(manga) {
-        const messageurl="https://reaperscans.com/livewire/message/";
-        let chapterList = [];
-        let requestdata = undefined;
-        let chapters = [];
-        //fetch first page for csrf-token, wireid and other informations
-        let url = new URL(manga.id, this.url);
-        let request = new Request(url, this.requestOptions);
-        let response = await fetch(request);
-        let responseText = await response.text();
-        //and chapters from first page
-        let parser = new DOMParser();
-        let doc = parser.parseFromString(responseText, 'text/html');
-        chapters = this.getChaptersFromDoc(doc);
-        chapterList.push(...chapters);
-        //build payload from first page data
-        requestdata = this.createPayloadForChapterPages(doc);
-        for (let page = 2, run = true; run; page++) {
-            chapters = [];
-            url = new URL(requestdata.payload.fingerprint.name, messageurl);
-            request = new Request(url, {
-                method: 'POST',
-                mode: 'cors',
-                referrer: requestdata.referrer,
-                referrerPolicy: 'strict-origin-when-cross-origin',
-                body: JSON.stringify(requestdata.payload),
-                credentials: 'include',
-                headers: {
-                    'Accept': 'text/html, application/xhtml+xml',
-                    "Content-Type": "application/json",
-                    "X-CSFRF-TOKEN": requestdata.token,
-                    "X-Livewire": "true",
-                    'x-referrer': requestdata.referrer,
-                }
-            });
-            response = await fetch(request);
-            let data = await response.json();
-            //data contains the html nodes with the chapters
-            //and the informations needed to request the next page
-            doc = parser.parseFromString(data.effects.html, 'text/html');
-            chapters = this.getChaptersFromDoc(doc);
-            chapters.length > 0 ? chapterList.push(...chapters) : run = false;
-            //update payload data using response data, for next fetch
-            requestdata.payload.serverMemo.checksum = data.serverMemo.checksum;
-            requestdata.payload.serverMemo.htmlHash = data.serverMemo.htmlHash;
-            requestdata.referrer = new URL(manga.id+"?page="+page, this.url).href;
-            requestdata.payload.serverMemo.data.page = page;
-            requestdata.payload.serverMemo.data.paginators.page = page;
-            requestdata.payload.updates[0].payload.params[0] = page+1;
+        const uri = new URL(manga.id, this.url);
+        const [ data ] = await this.fetchDOM(new Request(uri, this.requestOptions), 'main div[wire\\:id][wire\\:initial-data]');
```

Comment:
Use `data` instead of `[ data ]` ?

Reply:
Since fetchDOM returns an array, but we know there is only one element its more elegant than data[0] .
Nice interpolation trick i took for Ronny's code in Haruneko :)